### PR TITLE
[test] Move mem migrate to its own dir

### PR DESCRIPTION
**Pull request by hdelan**
Not having its own dir was breaking some builds.

### CODE REVIEW

File: test/conformance/CMakeLists.txt [hunk 1]
```diff
@@ -109,6 +109,7 @@ add_subdirectory(platform)
 add_subdirectory(device)
 add_subdirectory(context)
 add_subdirectory(memory)
+add_subdirectory(memory-migrate)
```
Comment (kbenzie) [inline test/conformance/CMakeLists.txt:112]: I think this should move down to line 140 along with all the other test suites that depend on kernel execution.

File: test/conformance/memory-migrate/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+if (UR_DPCXX)
```
Comment (kbenzie) [inline test/conformance/memory-migrate/CMakeLists.txt:1]: With the move of `add_subdirectory(memory-migrate)` this check will no longer be necessary.

Comment (kbenzie) [inline test/conformance/memory-migrate/CMakeLists.txt:1]: This is missing the license comment.

Comment (hdelan) [reply test/conformance/memory-migrate/CMakeLists.txt]: Thanks have added.

Comment (hdelan) [reply test/conformance/memory-migrate/CMakeLists.txt]: Thanks change made

File: test/conformance/CMakeLists.txt (hunk 1 above)
Comment (hdelan) [reply test/conformance/CMakeLists.txt]: OK sure, have moved.

**Review by kbenzie — APPROVED** [review]

**Pull request merged**